sympy/physics/vector/point.py

```diff
@@ -483,19 +483,46 @@ def vel(self, frame):
         Examples
         ========
 
-        >>> from sympy.physics.vector import Point, ReferenceFrame
+        >>> from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
         >>> N = ReferenceFrame('N')
         >>> p1 = Point('p1')
         >>> p1.set_vel(N, 10 * N.x)
         >>> p1.vel(N)
         10*N.x
+        >>> p = Point('p')
+        >>> q = dynamicsymbols('q')
+        >>> p.set_vel(N, 10 * N.x)
+        >>> p2 = Point('p2')
+        >>> p2.set_pos(p, q*N.x)
+        >>> p2.vel(N)
+        (Derivative(q(t), t) + 10)*N.x
 
         """
 
         _check_frame(frame)
         if not (frame in self._vel_dict):
-            raise ValueError('Velocity of point ' + self.name + ' has not been'
+            visited = []
+            queue = [self]
+            while queue: #BFS to find nearest point
+                node = queue.pop(0)
+                if node not in visited:
+                    visited.append(node)
+                    for neighbor, neighbor_pos in node._pos_dict.items():
+                        try:
+                            neighbor_pos.express(frame) #Checks if pos vector is valid
+                        except ValueError:
+                            continue
+                        try :
+                            neighbor_velocity = neighbor._vel_dict[frame] #Checks if point has its vel defined in req frame
+                        except KeyError:
+                            queue.append(neighbor)
+                            continue
+                        self._vel_dict[frame] =  self.pos_from(neighbor).dt(frame) + neighbor_velocity
```

Comment:
```suggestion
                        self._vel_dict[frame] =  self.pos_from(neighbor).dt(frame) + neighbor.vel(frame)
```

I think if the velocity in _vel_dict exists, then we should access it via the normal `.vel()` method.

Also, is there any reason not to use `self.set_vel()` here instead of `self._vel_dict[frame]`? It looks like `set_vel` is safe, as it only checks if the object is a frame.

Reply:
But, I see that `_check_frame` is also called at the top of this file. So, I guess this is doing the same thing.

Reply:
But doing it this way calls the function only once, as neigbhor_velocity is set in try statement

Reply:
The idea here is that calling `P.set_vel(N, ...)` versus `P._vel_dict[N] = ...` will ensure that it is set properly and reduce redudeant code. We should keep all code for setting velocities in `set_vel()`. Ideally, that is the only place that `P._vel_dict[] = ` is called.